src/connect/ConnectWizard.ts

```diff
@@ -202,8 +244,33 @@ export class ConnectWizard {
             activeItem: serviceChoices[0]
         });
         this._result.resourceName = pick.label;
+        // get the list of containers
+        const podNames = await kubectlClient.getPodNames(this._result.resourceName, this._result.targetNamespace);
+        if (podNames?.length === 0) {
```

Comment:
We should check for `(podNames === null || podNames.length === 0)` here.

If `podNames` is null, then `podNames?.length === undefined`, and `undefined !== 0`, so the `if` condition returns `false`.

Then later when you hit `podNames[0]` you'll get a runtime error `Cannot read properties of null`.

This stuff becomes really hard to reason about, which is why I'm really keen on checking for `<result> === null` _immediately_ after any assignment that might return null, and short-circuiting out of the function if that's an error condition.

Also, plug for an `errorable` type which avoids this confusion altogether...

Reply:
As discussed, ```errorable``` type will be another PR (refactor and type checking) but updated for now. Thanks for catching this.